src/cspviolationmonitor/CSPViolationMonitor.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Logger from '../logger/Logger';
+
+export default class CSPViolationMonitor {
+  private static loggers: Set<Logger> = new Set<Logger>();
+  private static shouldRegisterCSPMonitor: boolean = true;
+  private static added = false;
+
+  /* istanbul ignore next */
+  static registerCSPMonitor(logger?: Logger): void {
+    if (CSPViolationMonitor.shouldRegisterCSPMonitor) {
+      if (logger) {
+        CSPViolationMonitor.loggers.add(logger);
```

Comment:
I'm pretty sure I asked this in the last review: how do I remove a logger?

Also, you should probably move the `global` check up to the start of this function so we don't accrue a logger if we're going to bail out anyway.

Reply:
Oh you are saying we need to remove the logger in this CSPViolationMonitor. I thought you were asking we can remove the logger in the meetingSession, which is already existed. Sorry, I misunderstood it. Sure, I will remove the logger in the this CSPViolationMonitor component.

Reply:
I mean: once I call

```
registerCSPMonitor(myLogger);
```

how do I dispose of the logger? You need some method to unsubscribe.